src/VersionParser.php

```diff
@@ -216,7 +216,7 @@ public function normalizeBranch($name)
      *
      * @return LinkConstraintInterface
      */
-    public function parseConstraints($constraints)
+    public static function parseConstraints($constraints)
```

Comment:
Since all of the comparator methods and such are static, I figured we would probably want to make the public ones from VersionParser static too. If you disagree I can revert this change. I'm ok with either approach.